src/app/App.vue

```diff
@@ -99,8 +104,28 @@ const route = useRoute()
  */
 const routeKey = computed(() => route.path)
 const isWizard = computed(() => route.meta.isWizard === true)
-const shouldShowAppError = computed(() => store.getters.shouldShowAppError)
 const shouldShowOnboardingNotification = computed(() => store.getters.shouldShowOnboardingNotification)
+const loading = ref<boolean>(true)
+const error = ref<Error | undefined>()
+;(
+  async () => {
+    try {
+      await Promise.all([
+        // Fetches basic resources before setting up the router and mounting the
+        // application. This is mainly needed to properly redirect users to the
+        // onboarding flow in the appropriate scenarios.
+        store.dispatch('bootstrap'),
+        // Loads available policy types in order to populate the necessary
+        // information used for and policy lookups in the app.
+        store.dispatch('fetchPolicyTypes'),
+      ])
+    } catch (e) {
+      error.value = e as Error
+    } finally {
+      loading.value = false
+    }
+  }
+)()
```

Comment:
This is the stuff I moved into `App.vue`, now with loading and error integration.